stdlib/public/runtime/Errors.cpp

```diff
@@ -126,20 +133,20 @@ static void dumpStackTraceEntry(unsigned index, void *framePC) {
   // from the base address of where the image containing framePC is mapped.
   // This gives enough info to reconstruct identical debugging target after
   // this process terminates.
-  if (foundSymbol) {
-    static const char *backtraceEntryFormat = "%-4u %-34s 0x%0.16lx %s + %td\n";
-    fprintf(stderr, backtraceEntryFormat, index, libraryName.data(), symbolAddr,
-            symbolName.c_str(), ptrdiff_t(uintptr_t(framePC) - symbolAddr));
+  if (! shortOutput) {
+    static const char *moduleAndAddressFormat = "%-4u %-34s 0x%0.16lx ";
+    fprintf(stderr, moduleAndAddressFormat, index, libraryName.data(),
+            symbolAddr);
   } else {
-    static const char *backtraceEntryFormat = "%-4u %-34s 0x%0.16lx "
-                                              "<unavailable> + %td\n";
-    fprintf(stderr, backtraceEntryFormat, index, libraryName.data(),
-            uintptr_t(framePC),
-            ptrdiff_t(uintptr_t(framePC) - uintptr_t(syminfo.baseAddress)));
+    fprintf(stderr, "%s`", libraryName.data());
   }
-}
 
-#endif
+  fprintf(stderr, "%s + %td", symbolName.c_str(), offset);
+
+  if (! shortOutput) {
+    fprintf(stderr, "\n");
+  }
```

Comment:
It's best to make a single call to fprintf when reasonably possible to try to minimize problems with interleaving: stderr shouldn't be buffered in user space, so a single call is likely to result in a single call to write, which should minimize the amount of jumbling with other threads printing concurrently.  Of course there are limits to this — it would be too much to try to buffer the entire stack trace — but here I think it's pretty reasonably attainable.

On the other hand, it doesn't really work for your short format.